app/src/main/java/com/vitorpamplona/amethyst/ui/qrcode/QrCodeScanner.kt

```diff
@@ -1,156 +1,57 @@
-        analyzer
-    )
-
-    cameraProvider.bindToLifecycle(
-        lifecycleOwner,
-        selector,
-        imageAnalysis,
-        preview
-    )
-}
-
-class QRCodeAnalyzer(
-    private val onQrCodeScanned: (result: String?) -> Unit
-) : ImageAnalysis.Analyzer {
-
-    private val scanningOptions = BarcodeScannerOptions.Builder().setBarcodeFormats(Barcode.FORMAT_QR_CODE).build()
-
-    fun scanBarcodes(inputImage: InputImage) {
-        BarcodeScanning.getClient(scanningOptions).process(inputImage)
-            .addOnSuccessListener { barcodes ->
-                if (barcodes.isNotEmpty()) {
-                    onQrCodeScanned(barcodes[0].displayValue)
-                }
-            }
-            .addOnFailureListener {
-                it.printStackTrace()
-            }
-    }
-
-    @androidx.annotation.OptIn(androidx.camera.core.ExperimentalGetImage::class)
-    override fun analyze(imageProxy: ImageProxy) {
-        imageProxy.image?.let { image ->
-            val inputImage = InputImage.fromMediaImage(image, imageProxy.imageInfo.rotationDegrees)
-            scanBarcodes(inputImage)
-        }
-        imageProxy.close()
-    }
-}
+package com.vitorpamplona.amethyst.ui.qrcode
+
+import androidx.activity.compose.rememberLauncherForActivityResult
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.DisposableEffect
+import androidx.compose.ui.platform.LocalLifecycleOwner
+import androidx.compose.ui.res.stringResource
+import com.journeyapps.barcodescanner.ScanContract
+import com.journeyapps.barcodescanner.ScanOptions
+import com.vitorpamplona.amethyst.R
+import com.vitorpamplona.amethyst.service.nip19.Nip19
+
+@Composable
+fun QrCodeScanner(onScan: (String?) -> Unit) {
+    val lifecycleOwner = LocalLifecycleOwner.current
+
+    val parseQrResult = { it: String ->
+        try {
+            val nip19 = Nip19.uriToRoute(it)
+            val startingPage = when (nip19?.type) {
+                Nip19.Type.USER -> "User/${nip19.hex}"
+                Nip19.Type.NOTE -> "Note/${nip19.hex}"
+                else -> null
+            }
+
+            if (startingPage != null) {
+                onScan(startingPage)
+            } else {
+                onScan(null)
+            }
+        } catch (e: Throwable) {
+            // QR can be anything, do not throw errors.
+            onScan(null)
+        }
+    }
+
+    val qrLauncher =
+        rememberLauncherForActivityResult(ScanContract()) {
+            if (it.contents != null) {
+                parseQrResult(it.contents)
+            } else {
+                onScan(null)
+            }
+        }
+
+    val scanOptions = ScanOptions().apply {
+        setDesiredBarcodeFormats(ScanOptions.QR_CODE)
+        setPrompt(stringResource(id = R.string.point_to_the_qr_code))
+        setBeepEnabled(false)
+        setOrientationLocked(false)
+    }
+
+    DisposableEffect(lifecycleOwner) {
+        qrLauncher.launch(scanOptions)
+        onDispose { }
+    }

```

Comment:
This is the entirety of the new code needed to use zxing, and replaces all of the previous QR scanner code.